test/cluster/basic_test.go

```diff
@@ -101,3 +102,191 @@ func TestBasicCluster(t *testing.T) {
 		require.Equal(t, uint64(0), dataResp.Metadata.Version.TxNum)
 	}
 }
+
+// Scenario:
+// - start 3 servers in a cluster.
+// - wait for one to be the leader.
+// - stop one follower.
+// - submit a tx to the two remaining servers.
+// - start the stopped server.
+// - wait for one to be the new leader.
+// - make sure the stopped server is in sync with the transaction made while it was stopped.
+func TestNodeRecoveryFollower(t *testing.T) {
+	dir, err := ioutil.TempDir("", "int-test")
+	require.NoError(t, err)
+
+	nPort, pPort := getPorts(3)
+	setupConfig := &setup.Config{
+		NumberOfServers:     3,
+		TestDirAbsolutePath: dir,
+		BDBBinaryPath:       "../../bin/bdb",
+		CmdTimeout:          10 * time.Second,
+		BaseNodePort:        nPort,
+		BasePeerPort:        pPort,
+		CheckRedirectFunc: func(req *http.Request, via []*http.Request) error {
+			return errors.Errorf("Redirect blocked in test client: url: '%s', referrer: '%s', #via: %d", req.URL, req.Referer(), len(via))
+		},
+	}
+	c, err := setup.NewCluster(setupConfig)
+	require.NoError(t, err)
+
+	defer c.ShutdownAndCleanup()
+
+	require.NoError(t, c.Start())
+
+	leaderIndex := -1
+	require.Eventually(t, func() bool {
+		leaderIndex = c.AgreedLeader(t, 0, 1, 2)
+		return leaderIndex >= 0
+	}, 30*time.Second, 100*time.Millisecond)
+
+	keys := []string{"alice", "charlie", "dan"}
+	for _, key := range keys {
+		txID, rcpt, err := c.Servers[leaderIndex].WriteDataTx(t, worldstate.DefaultDBName, key, []byte(key+"-data"))
+		require.NoError(t, err)
+		if err != nil {
+			t.Logf("txID: %s, error: %s", txID, err)
+		} else {
```

Comment:
if there is an error, this is unreachable because of the require above. So you can drop the if else and just print:

`t.Logf("tx submitted: %s, %+v", txID, rcpt)`

Reply:
On the other hand, check the receipt and the validation info in order to make sure the TX is valid.

Reply:
done